### DISCUSSION TITLE
Load permission of only active role of user

### DISCUSSION THREAD

**Discussion by team-vardaam** [spatie/laravel-permission#1879] [Q&A]
Hello,

I have multiple roles for a user in my application. I need to develop a feature where logged in user can see his roles and can switch to another role. As package loads merged permissions of assigned roles, how could i load permission of only active role?

for marking a role active for user , i am thinking of adding an extra field to  `model_has_roles` table. switching to another role will make that role active for particular user. suggestions are greatly appreciated.

Thank you.

**Comment by erikn69** [comment]
Maybe you are mentioning the teams feature, where the user can change team and have different roles and permissions for each team
https://github.com/spatie/laravel-permission/pull/1804

**Reply by team-vardaam** [reply]
@erikn69 Thanks for answering. So if user have multiple roles like employee & accountant, will it create two teams (one for employee role, one for accountant ) ? if user switch to another role ( i.e. another team ) it will automatically load permissions, right? 

i think team contains multiple roles. so it may be possible that same role exists in another team with some other permission. which is not my use case here.

**Reply by erikn69** [reply]
> it will automatically load permissions, right?

Yes, I'm just trying to give you ideas, this package is not designed for that behavior, maybe there are other packages that work that way

>so it may be possible that same role exists in another team with some other permission

Yes, also there is global roles and they can be in all teams

**Comment by zulzayn** [comment]
@team-vardaam You can use the custom permission check to filter active role on each user. So first go to `app/Providers/AuthServiceProvider.php` and you can add like below in the boot method after  `$this->registerPolicies();` :

` Gate::before(function ($user, $ability) {
    return $user->hasActiveRolePermission($user, $ability) ?: null;
});`

Here hasActiveRolePermission is a custom method you need to implement yourself.

For this, inside user model `User.php` , you can do the logic to filter only active role, below lets assume that in table User you have column active_role_id with role_id as its value and relationship of activerole as one to one to Role model.

`  public static function hasActiveRolePermission($user, $ability){
        if ($user->activerole) {
            return $user->activerole->hasPermissionTo($ability);
        }
        else{
            return null;
        }
    }`

Finally, by default, register_permission_check_method is set to true in `config/permission.php`. Set this to false if you want to implement the above custom logic for checking permissions.
